### PR TITLE
Problem: K8s deployment with Access Token Authorization expects Access Token for GET calls

### PR DESCRIPTION
**Pull request by shahbazn**
## Description
K8s deployment with `secret-access-token` based authorization instead of three-scale should expect Secret-Access-Token header to be set only for POST calls. But currently it expects the same header for GET calls as well.

### CODE REVIEW

File: k8s/nginx-https/container/nginx.conf.template [hunk 1]
```diff
@@ -111,11 +112,21 @@ http {
...
       if ( $http_x_secret_access_token != "SECRET_ACCESS_TOKEN" ) {
+        set $auth_check 0;
+      }
+
+      # CODES: POST=1 and GET=0
+      if ( $request_method = POST ) {
+        set $auth_check "${auth_check}1";
+      }
+
+      # Block POST requests with invalid Secrete Access Token
```
Comment (muawiakh) [inline k8s/nginx-https/container/nginx.conf.template]: Secret*

**Comment by shahbazn** [comment]
@muawiakh fixed

**Comment by ttmc** [comment]
Sigh, I don't understand NGINX, I mean could they be any more arcane? But if this works then good.

**Comment by ttmc** [comment]
Maybe you could add some comments for future readers to understand what is happening, maybe with links to relevant documentation online.

File: k8s/nginx-https/container/nginx.conf.template [hunk 2]
```diff
@@ -111,11 +112,21 @@ http {
 
       # Check for security header to authorize POST requests
       if ( $http_x_secret_access_token != "SECRET_ACCESS_TOKEN" ) {
+        set $auth_check 0;
+      }
+
+      # CODES: POST=1 and GET=0
```
Comment (muawiakh) [inline k8s/nginx-https/container/nginx.conf.template]: You are not getting for GET-> 0 anywhere, might as well remove it.

**Comment by shahbazn** [comment]
@muawiakh for three scale there is a separate conf file `nginx.conf.threescale.template` this file will only we used if you have set `AUTHORIZATION_MODE ` to `secrete-token`

**Comment by shahbazn** [comment]
closing this in favor of https://github.com/bigchaindb/bigchaindb/pull/2134#pullrequestreview-103934212

**Pull request closed without merging**